lib/WTSI/NPG/HTS/BatchPublisher.pm

```diff
@@ -224,6 +231,27 @@ has 'require_checksum_cache' =>
         if (!$mlwh_json_cb->($obj, $dest_coll, $filename)){
           $self->logcroak("Failed to add '$file' to ml warehouse json file");
         }
+        if ($self->mlwh_locations){
+          my $target;
+          my $pid;
+          for my $avu (@primary_avus){
+            if ($avu->{attribute} eq 'target'){
+              $target = $avu->{value};
+            }elsif ($avu->{attribute} eq 'id_product'){
+              $pid = $avu->{value};
+            }
+            if (defined($target) && $pid){
+              last;
+            }
+          }
```

Comment:
It would be good to have some comments on what's happening here. Also, AFAIR the mlwh_json_cb is still in place for the other code that needs it. It would also be helpful to have some comments around that saying that it's deprecated and will be removed. That will help anyone else who comes along and wonders why there's currently two ways of doing this.

Reply:
Thanks - those are helpful.